examples/workflows/cockroach_basic_cluster/main.tf

```diff
@@ -0,0 +1,87 @@
+variable "cluster_name" {
+  type     = string
+  nullable = false
+}
+
+variable "sql_user_name" {
+  type     = string
+  nullable = false
+  default  = "maxroach"
+}
+
+# Remember that even variables marked sensitive will show up
+# in the Terraform state file. Always follow best practices
+# when managing sensitive info.
+# https://developer.hashicorp.com/terraform/tutorials/configuration-language/sensitive-variables#sensitive-values-in-state
+variable "sql_user_password" {
+  type      = string
+  nullable  = false
+  sensitive = true
+}
+
+variable "cloud_provider" {
+  type     = string
+  nullable = false
+  default  = "GCP"
+}
+
+variable "cloud_provider_regions" {
+  type     = list(string)
+  nullable = false
+  default  = ["us-central1"]
+}
+
+terraform {
+  required_providers {
+    cockroach = {
+      source = "cockroachdb/cockroach"
+    }
+  }
+}
+provider "cockroach" {
+  # export COCKROACH_API_KEY with the cockroach cloud API Key
+}
+
+resource "cockroach_cluster" "example" {
+  name           = var.cluster_name
+  cloud_provider = var.cloud_provider
+  plan           = "BASIC"
+  serverless     = {}
+  regions        = [for r in var.cloud_provider_regions : { name = r }]
+}
+
+resource "cockroach_sql_user" "example" {
+  name       = var.sql_user_name
+  password   = var.sql_user_password
+  cluster_id = cockroach_cluster.example.id
+}
+
+resource "cockroach_database" "example" {
+  name       = "example-database"
+  cluster_id = cockroach_cluster.example.id
+}
+
+resource "cockroach_service_account" "example_scoped_sa" {
+  name        = "example-scoped-service-account"
+  description = "A service account providing limited read access to single cluster."
+}
+
+resource "cockroach_user_role_grant" "example_limited_access_scoped_grant" {
```

Comment:
Perhaps we can change this to something that is more widely used, like Cluster Admin? I don't think Cluster Operator is broadly used or something we recommend. If we want to show how to grant no privilege at all, we could grant ORG MEMBER. But I think that's granted by default.

Reply:
This is how the other examples work: it would be a separate change to do something like that, since it's not connected with Cloud 2.0. You can file an issue and/or talk with @marksoper, since a change like this should cut across all the cluster types.

Reply:
@mdlinville I agree with @andy-kimball that the two concerns: 1. how we use role grants in our examples and 2. our development for Cloud 2.0 should be separated.

The role grant concern you're referring to was introduced ~4 months ago when @fantapop added support for role grants in part because of an ask from JPMC.

Reply:
I understood the use of CLUSTER_OPERATOR was meant to show the assignment of cluster management capability to a single user for that cluster alone - which was part of the original JPMC requirement.

Reply:
@mdlinville could you please clarify the concern with CLUSTER_OPERATOR. From this doc it looks like CLUSTER_OPERATOR is consistent with our current docs and is less permissive than CLUSTER_ADMIN while still giving the user cluster management capabilities.
https://www.cockroachlabs.com/docs/cockroachcloud/authorization

Reply:
I suppose I was just remembering discussions during development of FGAC, where I was asked to use Cluster Admin rather than Cluster Operator in some contexts. I think I'm good here.